docs/4.4/api-reference.md

```diff
@@ -11,57 +11,328 @@ Teleport is currently working on documenting our API.
+  Roles: teleport.Roles{teleport.RoleProxy},
+  TTL:   time.Hour,
+})
+if err != nil {
+  return err
+}
+```
+
+**Update token**
+
+```go
+// updates the token to be a proxy token
+token.SetRoles(teleport.Roles{teleport.RoleProxy})
+if err := client.UpsertToken(token); err != nil {
+  return err
+}
+```
+
+**Delete token**
+
+```go
+if err := client.DeleteToken(tokenString); err != nil {
+  return err
+}
+```
+
+## Cluster Labels
+
+The root cluster in a [trusted cluster](admin-guide.md#adding-nodes-to-the-cluster) can add/update cluster labels on leaf clusters.
+
+**Create a leaf cluster token with Labels**
+
+Leaf clusters will inherit the labels of the [join token](trustedclusters.md#join-tokens) used to add them to the trusted cluster. This is the preferred method of managing cluster tokens.
+
+Follow the trusted cluster join token [docs](trustedclusters.md#join-tokens) to use this token to create a leaf cluster using `tctl`.
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // You can provide 'Token' for a static token name
+  Roles: teleport.Roles{teleport.RoleTrustedCluster},
+  TTL:   time.Hour,
+  Labels: map[string]string{
+    "env": "staging",
+  },
+})
+```
+
+**Update leaf cluster's labels**
+
+You can also update a leaf cluster's labels from the root cluster using the auth api if necessary.
+
+```go
+rc, err := client.GetRemoteCluster("leafClusterName")
+if err != nil {
+  return err
+}
+
+md := rc.GetMetadata()
+md.Labels = map[string]string{"env": "prod"}
+rc.SetMetadata(md)
+
+if err = client.UpdateRemoteCluster(ctx, rc); err != nil {
+  return err
+}
+
+```
+
+## Access Workflow
+
+[Access Workflow](enterprise/workflow/index.md) can be used to dynamically approve and deny a user access to local or remote resources in a trusted cluster.
+
+**Retrieve access requests**
+
+```go
+filter := services.AccessRequestFilter{State: services.RequestState_PENDING}
+ars, err := client.GetAccessRequests(ctx, filter)
+if err != nil {
+  return err
+}
+```
+
+**Create access request**
+
+```go
+// create a new access request for api-admin to use the admin role in the cluster
+ar, err := services.NewAccessRequest("api-admin", "admin")
+if err != nil {
+  return err
+}
+
+if err = client.CreateAccessRequest(ctx, accessReq); err != nil {
+  return err
+}
+```
+
+**Approve access request**
+
+```go
+aruApprove := services.AccessRequestUpdate{
+  RequestID: accessReqID,
+  State:     services.RequestState_APPROVED,
+}
+if err := client.SetAccessRequestState(ctx, aruApprove); err != nil {
+  return err
+}
+```
+
+**Deny access request**
+
+```go
+aruDeny := services.AccessRequestUpdate{
+  RequestID: accessReqID,
+  State:     services.RequestState_DENIED,
+}
+if err := client.SetAccessRequestState(ctx, aruDeny); err != nil {
+  return err
+}
+```
+
+**Delete access request**
+
+```go
+if err = client.DeleteAccessRequest(ctx, accessReqID); err != nil {
```

Comment:
This should be `if err := ...` the above snippet will only work if you have err defined already.